external-crates/move/crates/move-compiler/tests/move_2024/typing/index_syntax_methods_miscalled.move

```diff
@@ -0,0 +1,161 @@
+#[defines_primitive(vector)]
+module std::vector {
+    #[bytecode_instruction]
+    native public fun empty<Element>(): vector<Element>;
+
+    #[bytecode_instruction]
+    #[syntax(index)]
+    native public fun borrow<Element>(v: &vector<Element>, i: u64): &Element;
+
+    #[bytecode_instruction]
+    #[syntax(index)]
+    native public fun borrow_mut<Element>(v: &mut vector<Element>, i: u64): &mut Element;
+}
+
+module a::s {
+
+    public struct S has drop { t: vector<u64> }
+
+    #[syntax(index)]
+    public fun borrow_s(s: &S, i: u64): &u64 {
+        &s.t[i]
+    }
+
+    #[syntax(index)]
+    public fun borrow_s_mut(s: &mut S, i: u64): &mut u64 {
+        &mut s.t[i]
+    }
+
+    public fun miscall0(s: &S, i: u32): &u64 {
+        &s.t[i]
+    }
+
+    public fun miscall1(s: &mut S, i: u32): &mut u64 {
+        &mut s.t[i]
+    }
+
+    public fun miscall2<T>(s: &S, i: T): &u64 {
+        &s.t[i]
+    }
+
+
+    public fun miscall3<T>(s: &mut S, i: T): &mut u64 {
+        &mut s.t[i]
+    }
```

Comment:
Are these supposed to be the same as the others above?

Reply:
No, this explicitly miscalls with a `T` instead of a `u64`